### PR TITLE
fix(gatsby-plugin-image): Make onLoad callback work on first load

### PR DESCRIPTION
**Pull request by tyhopp**
## Description

Fixes `onLoad` callback execution on first load. Also moves `onStartLoad` in this scenario to execute at the right time (not in the load event listener callback).

### Documentation

N/A

## Related Issues

Fixes https://github.com/gatsbyjs/gatsby/issues/36373

[sc-54332]

### CODE REVIEW

File: packages/gatsby-plugin-image/src/components/gatsby-image.browser.tsx [hunk 1]
```diff
@@ -122,12 +122,13 @@ const GatsbyImageHydrator: FC<GatsbyImageProps> = function GatsbyImageHydrator({
           ssrImage.removeAttribute(`data-gatsby-image-ssr`)
         }, 0)
       } else {
-        document.addEventListener(`load`, function onLoadListener() {
-          document.removeEventListener(`load`, onLoadListener)
+        onStartLoad?.({
+          wasCached: true,
+        })
+
+        ssrImage.addEventListener(`load`, function onLoadListener() {
```
Comment (LekoArts) [inline packages/gatsby-plugin-image/src/components/gatsby-image.browser.tsx:129]: I'm assuming we used `document` here because `ssrImage` would be falsy in this else block. Was this change necessary to make things work? If yes, we should probably check for truthy `ssrImage` before using it

Comment (tyhopp) [reply packages/gatsby-plugin-image/src/components/gatsby-image.browser.tsx:129]: I think you might be looking at the outer `if` condition, this `else` corresponds to `if (ssrImage.complete) { ... }`. 

That was the main change to make it work, I'm not quite sure why it was `document` to begin with either (if we listen on `document` we would need to inspect the event target to differentiate for it to work that way anyway)

Comment (LekoArts) [reply packages/gatsby-plugin-image/src/components/gatsby-image.browser.tsx:129]: oh, lol, yeah you're right. Missed that line 111

**Review by LekoArts — APPROVED** [review]
